tests/Functional/ContentOrganisationRenderTest.php

```diff
@@ -0,0 +1,320 @@
+
+    // Assert navigation part.
+    $navigation = $this->assertSession()->elementExists('css', 'nav.ecl-inpage-navigation');
+    $assert = new InPageNavigationAssert();
+    $expected_values = [
+      'title' => 'Page contents',
+      'list' => [
+        ['label' => 'Description', 'href' => '#description'],
+        ['label' => 'Contact', 'href' => '#contact'],
+      ],
+    ];
+    $assert->assertPattern($expected_values, $navigation->getOuterHtml());
+
+    $logo = $this->assertSession()->elementExists('css', '.ecl-col-lg-3 img.ecl-media-container__media');
+    $this->assertContains('styles/oe_theme_ratio_3_2_medium/public/example_1.jpeg', $logo->getAttribute('src'));
+    $this->assertEquals('Alt', $logo->getAttribute('alt'));
+
+    // Assert content part.
+    $wrapper = $this->assertSession()->elementExists('css', '.ecl-row.ecl-u-mt-l');
+    $content = $this->assertSession()->elementExists('css', '.ecl-col-lg-9', $wrapper);
+    $this->assertSession()->elementsCount('css', '.ecl-col-lg-9', 1);
+    $content_items = $content->findAll('xpath', '/div');
+    $this->assertCount(2, $content_items);
+
+    // Assert header of first field group.
+    $this->assertContentHeader($content_items[0], 'Description', 'description');
+
+    // Assert values for first group.
+    $body = $content_items[0]->findAll('css', '.ecl-editor');
+    $this->assertCount(1, $body);
+    $this->assertEquals('My body text', $body[0]->getText());
+
+    // Assert header of second field group.
+    $this->assertContentHeader($content_items[1], 'Contact', 'contact');
+
+    // Assert Organisation's contacts.
+    $contact_headers = $content_items[1]->findAll('css', 'h2');
+    $this->assertEquals($contact_headers[0]->getText(), 'Contact');
+    $this->assertContactRendering($content_items[1], 'general_contact');
+  }
+
+  /**
+   * Asserts field group header.
+   *
+   * @param \Behat\Mink\Element\NodeElement $element
+   *   Field group content.
+   * @param string $title
+   *   Expected title.
+   * @param string $id
+   *   Expected id.
+   */
+  protected function assertContentHeader(NodeElement $element, string $title, string $id): void {
+    $header = $element->find('css', 'h2.ecl-u-type-heading-2');
+    $this->assertEquals($title, $header->getText());
+    $this->assertEquals($id, $header->getAttribute('id'));
+  }
+
+  /**
+   * Creates Contact entity.
+   *
+   * @param string $name
+   *   Entity name. Is used as a parameter for test data.
+   * @param string $bundle
+   *   Entity bundle.
+   * @param int $status
+   *   Entity status.
+   *
+   * @return \Drupal\oe_content_entity_contact\Entity\ContactInterface
+   *   Contact entity.
+   */
+  protected function createContactEntity(string $name, string $bundle, int $status): ContactInterface {
+    $file = $this->createFile(drupal_get_path('theme', 'oe_theme') . '/tests/fixtures/placeholder.png');
+    $media = $this->createMediaImage($file, [
```

Comment:
Same holds for this then. We will have a dedicated ticket to refactor and better organize tests. Let's roll back to the previous, yet verbose, method.

Reply:
done